README
Paludis version 0.01
====================

The README is used to introduce the module and provide instructions on
how to install the module, any machine dependencies it may have (for
example C compilers and installed libraries) and any other information
that should be provided before the module is installed.

A README file is required for CPAN modules since CPAN extracts the
README file from a module distribution so that people browsing the
archive can use it get an idea of the modules uses. It is usually a
good idea to provide version information here so that people can
decide whether fixes for the module are worth downloading.

INSTALLATION

To install this module type the following:

   perl Makefile.PL
   make
   make test
   make install

DEPENDENCIES

This module requires these other modules and libraries:

  blah blah blah

COPYRIGHT AND LICENCE

Put the correct copyright and licence information here.

Copyright (C) 2008 by Mike Kelly

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself, either Perl version 5.8.8 or,
at your option, any later version of Perl 5 you may have available.




Changes
Revision history for Perl extension Paludis.

0.01  Thu Jul  3 02:48:29 2008
	- original version; created by h2xs 1.23 with options
		-A -n Paludis



VersionSpec/Makefile.PL
use 5.008008;
use ExtUtils::MakeMaker;

$CC = 'g++';

# See lib/ExtUtils/MakeMaker.pm for details of how to influence
# the contents of the Makefile that is written.
my $paludis_include_dir = `pkg-config --cflags-only-I paludis`;
WriteMakefile(
    NAME              => 'Paludis::VersionSpec',
    VERSION_FROM      => 'lib/Paludis/VersionSpec.pm',
    PREREQ_PM         => {},
    ABSTRACT_FROM     => 'lib/Paludis/VersionSpec.pm',
    AUTHOR            => 'Mike Kelly <pioto@pioto.org>',
    LIBS              => [split / /, `pkg-config --libs-only-l paludis`],
    DEFINE            => '',
    CC                => $CC,
    CCFLAGS           => `pkg-config --cflags-only-other paludis`,
    LD                => '$(CC)',
    INC               => "-I. -I.. $paludis_include_dir",
	# Un-comment this if you add C files to link with later:
    # OBJECT            => '$(O_FILES)', # link all the C files too
    XSOPT             => '-C++',
    TYPEMAPS          => ['../perlobject.map'],
);


VersionSpec/lib/Paludis/VersionSpec.pm
package Paludis::VersionSpec;

use 5.008008;
use strict;
use warnings;

require Exporter;

our @ISA = qw(Exporter);

# Items to export into callers namespace by default. Note: do not export
# names by default without a very good reason. Use EXPORT_OK instead.
# Do not simply export all your public functions/methods/constants.

# This allows declaration	use Paludis ':all';
# If you do not need this, moving things directly into @EXPORT or @EXPORT_OK
# will save memory.
our %EXPORT_TAGS = ( 'all' => [ qw(
	
) ] );

our @EXPORT_OK = ( @{ $EXPORT_TAGS{'all'} } );

our @EXPORT = qw(
	
);

our $VERSION = '0.01';

require XSLoader;
XSLoader::load('Paludis::VersionSpec', $VERSION);

# Preloaded methods go here.

use overload
    '<=>'    => \&Paludis::VersionSpec::comp,
    fallback  => 1;

1;
__END__
# Below is stub documentation for your module. You'd better edit it!

=head1 NAME

Paludis::VersionSpec - Compare different versions according to Paludis'
rules

=head1 SYNOPSIS

  use Paludis;

  my $ver0 = Paludis::VersionSpec->new("0");
  my $ver0r0 = Paludis::VersionSpec->new("0-r0");

  print "0 == 0-r0\n" if $ver0 == $ver0r0;

=head1 DESCRIPTION

The Paludis package provides bindings for Paludis, the Other Package
Mangler. Paludis is a multi-format package manager. It can currently be
used on Gentoo and derivatives, using the ebuild and VDB repository
formats, and on Exherbo using the exheres repository format.

At the moment, these bindings simply provide access the VersionSpec
object (which allows comparing different version strings).

=head2 EXPORT

None by default.



=head1 SEE ALSO

L<Paludis>

man paludis

paludis-sekrit@lists.pioto.org
paludis-users@lists.pioto.org

L<http://paludis.pioto.org/>

=head1 AUTHOR

Mike Kelly, E<lt>pioto@pioto.orgE<gt>

=head1 COPYRIGHT AND LICENSE

Copyright (C) 2008 by Mike Kelly

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself, either Perl version 5.8.8 or,
at your option, any later version of Perl 5 you may have available.


=cut


lib/Paludis.pm
package Paludis;

use 5.008008;
use strict;
use warnings;
use Carp;

use Paludis::VersionSpec;

require Exporter;
use AutoLoader;

our @ISA = qw(Exporter);

# Items to export into callers namespace by default. Note: do not export
# names by default without a very good reason. Use EXPORT_OK instead.
# Do not simply export all your public functions/methods/constants.

# This allows declaration	use Paludis ':all';
# If you do not need this, moving things directly into @EXPORT or @EXPORT_OK
# will save memory.
our %EXPORT_TAGS = ( 'all' => [ qw(
    PALUDIS_VERSION_MAJOR
    PALUDIS_VERSION_MINOR
    PALUDIS_VERSION_MICRO
) ] );

our @EXPORT_OK = ( @{ $EXPORT_TAGS{'all'} } );

our @EXPORT = qw(
    PALUDIS_VERSION_MAJOR
    PALUDIS_VERSION_MINOR
    PALUDIS_VERSION_MICRO
);

sub AUTOLOAD {
    # This AUTOLOAD is used to 'autoload' constants from the constant()
    # XS function.

    my $constname;
    our $AUTOLOAD;
    ($constname = $AUTOLOAD) =~ s/.*:://;
    croak "&Bob::constant not defined" if $constname eq 'constant';
    my ($error, $val) = constant($constname);
    if ($error) { croak $error; }
    {
    no strict 'refs';
    *$AUTOLOAD = sub { $val };
    }
    goto &$AUTOLOAD;
}

our $VERSION = '0.01';

require XSLoader;
XSLoader::load('Paludis', $VERSION);

# Preloaded methods go here.

1;
__END__

=head1 NAME

Paludis - Perl bindings for Paludis, the Other Package Mangler

=head1 SYNOPSIS

  use Paludis ':all';

  print PALUDIS_VERSION_MAJOR, ".", PALUDIS_VERSION_MINOR, ".", PALUDIS_VERSION_MICRO, "\n";

=head1 DESCRIPTION

The Paludis package provides bindings for Paludis, the Other Package
Mangler. Paludis is a multi-format package manager. It can currently be
used on Gentoo and derivatives, using the ebuild and VDB repository
formats, and on Exherbo using the exheres repository format.

At the moment, these bindings simply provide access to the
PALUDIS_VERSION constants (defining the version of paludis against which
the module was built), and to the VersionSpec object (which allows
comparing different version strings).

=head2 EXPORT

None by default.

=head2 Exportable constants

  PALUDIS_VERSION_MAJOR
  PALUDIS_VERSION_MINOR
  PALUDIS_VERSION_MICRO

=head1 SEE ALSO

L<Paludis::VersionSpec>

man paludis

paludis-sekrit@lists.pioto.org
paludis-users@lists.pioto.org

L<http://paludis.pioto.org/>

=head1 AUTHOR

Mike Kelly, E<lt>pioto@pioto.orgE<gt>

=head1 COPYRIGHT AND LICENSE

Copyright (C) 2008 by Mike Kelly

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself, either Perl version 5.8.8 or,
at your option, any later version of Perl 5 you may have available.

=cut
